Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewLayout.cs

```diff
@@ -67,24 +66,32 @@ protected virtual void HandlePropertyChanged(PropertyChangedEventArgs  propertyC
 
 		public abstract void ConstrainTo(CGSize size);
 
+		[Export("collectionView:willDisplayCell:forItemAtIndexPath:")]
+		public virtual void WillDisplayCell(UICollectionView collectionView, UICollectionViewCell cell, NSIndexPath path)
+		{
+			if (_needCellSizeUpdate)
+			{
+				// Our cell size/estimate is out of date, probably because we moved from zero to one item; update it
+				_needCellSizeUpdate = false;
+				DetermineCellSize();
+			}
+		}
+
 		[Export("collectionView:layout:insetForSectionAtIndex:")]
-		[CompilerGenerated]
```

Comment:
there's a history behind those... probably a way to make sure r# wasn't raising warnings...

Reply:
Doesn't seem to bother R#er to leave them out. At least not with the latest version.